File: java/NCHU/src/Attendance.java
import java.io.BufferedReader;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.UnsupportedEncodingException;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLEncoder;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import org.json.JSONException;
import org.json.JSONObject;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

public class Attendance {
	private final String Web_login = "http://psf.nchu.edu.tw/punch/login_chk.jsp";
	private final String Web_save = "http://psf.nchu.edu.tw/punch/PunchTbl_save";
	private final String Web_record = "http://psf.nchu.edu.tw/punch/PunchTbl_Q";
	private final String Web_print = "http://psf.nchu.edu.tw/punch/FormPunch_A.jsp";
	private final String Web_schno = "http://psf.nchu.edu.tw/punch/ComeQueryPunch_A.jsp";
	private String JSESSIONID= null;
	
	public Attendance(){

	}
	
	public ArrayList GetSchno(String jessionID) throws IOException{
		URL url = new URL(Web_schno);
		HttpURLConnection http = (HttpURLConnection) url.openConnection();
		
		http.setRequestProperty("Cookie", "JSESSIONID=" + jessionID);
		http.setRequestMethod("GET");

        int responseCode = http.getResponseCode();
        StringBuffer response = new StringBuffer();
        
		BufferedReader in = new BufferedReader(new InputStreamReader(http.getInputStream()));
		String inputLine;
		
		while ((inputLine = in.readLine()) != null) {
			response.append(inputLine);
		}
		in.close();  		
        
        Document doc = Jsoup.parse(response.toString(), "UTF-8");
        ArrayList<String> data = new ArrayList<String>();
        Elements schnos = doc.getElementsByTag("option");
        for(Element schno : schnos ){
        	data.add(schno.val());
        }

        //System.out.println(data.toString().substring(0,data.length()-1));
        
		return data;
	} 
	
	public String PrintRecord(String schno ,String dateBeg ,String dateEnd ,String jessionID) throws IOException{
		URL url = new URL(Web_print+"?schno="+schno+"&dtQryBeg="+dateBeg+"&dtQryEnd="+dateEnd);
		HttpURLConnection http = (HttpURLConnection) url.openConnection();
		
		http.setRequestProperty("Cookie", "JSESSIONID=" + jessionID);
		http.setRequestMethod("GET");
		
        int responseCode = http.getResponseCode();
        StringBuffer response = new StringBuffer();
        
    	BufferedReader in = new BufferedReader(new InputStreamReader(http.getInputStream(),"UTF-8"));
		String inputLine;
		
		while ((inputLine = in.readLine()) != null) {
			response.append(inputLine);
		}
		in.close();  		

        //System.out.println(response.toString());
        return response.toString();
	}
	
	public void AddRecord(String date, String work ,String schno ,String jessionID) throws IOException{
		URL url = new URL(Web_save);
		HttpURLConnection http = (HttpURLConnection) url.openConnection();
		
		//設定Cookie
		http.setRequestProperty("Cookie", "JSESSIONID=" + jessionID);
		
		//設定Request為POST
		http.setRequestMethod("POST");
		http.setDoOutput(true);

		String data = "insert=true&date=" + date + "&work=" + work + "&schno=" + schno;
		DataOutputStream wr = new DataOutputStream(http.getOutputStream());
		wr.write(data.getBytes("utf-8"));
		wr.close();

		http.getResponseCode();
	}
	
	public void DelRecord(String rid ,String jessionID) throws IOException{
		URL url = new URL(Web_save);
		HttpURLConnection http = (HttpURLConnection) url.openConnection();
		
		//設定Cookie
		http.setRequestProperty("Cookie", "JSESSIONID=" + jessionID);
		http.setUseCaches (true);
		
		//設定Request為POST
		http.setRequestMethod("POST");
		http.setDoOutput(true);
		
		String data = "delete=true&rid="+URLEncoder.encode(rid, "utf-8");

		DataOutputStream wr = new DataOutputStream(http.getOutputStream());
		wr.write(data.getBytes("utf-8"));
		wr.close();

		http.getResponseCode();
	}
	
	//取得出勤紀錄(Json)
	public String GetRecord(String user,String jessionID) throws IOException{
		URL url = new URL(Web_record+"?acc="+user);
		HttpURLConnection http = (HttpURLConnection) url.openConnection();
		
		http.setRequestProperty("Cookie", "JSESSIONID=" + jessionID);
		http.setRequestMethod("GET");
		
        int responseCode = http.getResponseCode();
        StringBuffer response = new StringBuffer();
        
		BufferedReader in = new BufferedReader(new InputStreamReader(http.getInputStream(),"UTF-8"));
		String inputLine;
		
		while ((inputLine = in.readLine()) != null) {
			response.append(inputLine);
		}
		in.close();  		

        return response.toString();
	}

	
	public String GetJSESSIONID(){
		return JSESSIONID;
	}
	
	//Login 並取得JSESSIONID
	public boolean Login(String user,String pwd) throws IOException{
		URL url = new URL(Web_login);
		HttpURLConnection http = (HttpURLConnection) url.openConnection();
		
		http.setRequestMethod("POST");
		http.setDoOutput(true);

		String data = "txtLoginID=" + user + "&txtLoginPWD=" + pwd;
		DataOutputStream wr = new DataOutputStream(http.getOutputStream());
		wr.write(data.getBytes("utf-8"));
		wr.close();
		
		Map<String, List<String>> requestCookie = http.getHeaderFields();
		//System.out.println(requestCookie);
		
		for (int i = 0; i < requestCookie.get("Set-Cookie").size(); i++) {
			if (requestCookie.get("Set-Cookie").get(i).contains("JSESSIONID")) {
				JSESSIONID = requestCookie.get("Set-Cookie").get(i).split("=")[1].split(";")[0];
			}
		}
		http.getResponseCode();
		
		//接收網頁標籤資訊，確認是否登入成功
		BufferedReader in = new BufferedReader(new InputStreamReader(http.getInputStream()));
		String inputLine;
		StringBuffer response = new StringBuffer();
		
		while ((inputLine = in.readLine()) != null) {
			response.append(inputLine);
		}
		in.close();
		
		//System.out.println(response.toString());
		
		if (response.toString().contains("Menu")) {
			return true;
		}
		return false;
	}
}

File: php/attendance.php
<?PHP
include_once('simple_html_dom.php');

class Attendance 
{
	private $Web_login = "http://psf.nchu.edu.tw/punch/login_chk.jsp";
	private $Web_save = "http://psf.nchu.edu.tw/punch/PunchTbl_save";
	private $Web_record = "http://psf.nchu.edu.tw/punch/PunchTbl_Q";
	private $Web_print = "http://psf.nchu.edu.tw/punch/FormPunch_A.jsp";
	private $Web_schno = "http://psf.nchu.edu.tw/punch/ComeQueryPunch_A.jsp";
	private $JSESSIONID= null;

	/* 
	 * * * * * * * * * * * DelRecord * * * * * * * * * * * 
	 */
	function DelRecord($rid, $jessionID){
		$PostData = "delete=true&rid=".$rid;

		$ch = curl_init();
		// setting post url
		curl_setopt($ch, CURLOPT_URL, $this->Web_save);

		// use post mode
		curl_setopt( $ch, CURLOPT_POST, 1);
		
		// return web content
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);

		// post data content
		curl_setopt($ch, CURLOPT_POSTFIELDS, $PostData);

		// setting cookie
		curl_setopt($ch, CURLOPT_COOKIE, "JSESSIONID=".$jessionID);

		$result = curl_exec($ch);
	}

	/* 
	 * * * * * * * * * * * AddRecord * * * * * * * * * * * 
	 */
	function AddRecord($date, $work, $schno, $jessionID){
		$PostData = "insert=true&date=".$date."&work=".$work."&schno=".$schno;

		$ch = curl_init();
		// setting post url
		curl_setopt($ch, CURLOPT_URL, $this->Web_save);

		// use post mode
		curl_setopt( $ch, CURLOPT_POST, 1);
		
		// return web content
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);

		// post data content
		curl_setopt($ch, CURLOPT_POSTFIELDS, $PostData);

		// setting cookie
		curl_setopt($ch, CURLOPT_COOKIE, "JSESSIONID=".$jessionID);

		$result = curl_exec($ch);
	}

	/* 
	 * * * * * * * * * * * PrintRecord * * * * * * * * * * * 
	 */
	function PrintRecord($schno, $dateBeg, $dateEnd, $jessionID){
		$GetData = "?schno=".$schno."&dtQryBeg=".$dateBeg."&dtQryEnd=".$dateEnd;

		$ch = curl_init();
		
		// setting post url
		curl_setopt($ch, CURLOPT_URL, $this->Web_print.$GetData);
		
		// return web content
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
		
		// setting cookie
		curl_setopt($ch, CURLOPT_COOKIE, "JSESSIONID=".$jessionID);

		$result = curl_exec($ch);

		return $result;
	}

	/* 
	 * * * * * * * * * * * GetRecord * * * * * * * * * * * 
	 */
	function GetRecord($user, $jessionID){
		$GetData = "?acc=".$user;
		
		$ch = curl_init();
		
		// setting post url
		curl_setopt($ch, CURLOPT_URL, $this->Web_record.$GetData);
		
		// return web content
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
		
		// setting cookie
		curl_setopt($ch, CURLOPT_COOKIE, "JSESSIONID=".$jessionID);

		$result = curl_exec($ch);

		return $result;

	}

	/* 
	 * * * * * * * * * * * GetSchno * * * * * * * * * * * 
	 */
	function GetSchno($jessionID){
		$data = array();

		$ch = curl_init();
		
		// setting post url
		curl_setopt($ch, CURLOPT_URL, $this->Web_schno);
		
		// return web content
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
		
		// setting cookie
		curl_setopt($ch, CURLOPT_COOKIE, "JSESSIONID=".$jessionID);

		$result = curl_exec($ch);

		$Htmlparser = str_get_html($result);
		foreach($Htmlparser->find('option') as $element) 
       		array_push($data,$element->value);
		return $data;
	}


	/* 
	 * * * * * * * * * * * GetJSESSIONID * * * * * * * * * * * 
	 */

	function GetJSESSIONID(){
		return $this->JSESSIONID;
	}

	/* 
	 * * * * * * * * * * * Login * * * * * * * * * * * 
	 */
	function Login($user,$pwd){
		$PostData = "txtLoginID=".$user."&txtLoginPWD=".$pwd;

		$ch = curl_init();
		// setting post url
		curl_setopt($ch, CURLOPT_URL, $this->Web_login);

		// use post mode
		curl_setopt( $ch, CURLOPT_POST, 1);
		
		// return web content
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
		
		// get web request header
		curl_setopt( $ch, CURLOPT_HEADER, 1);

		// post data content
		curl_setopt($ch, CURLOPT_POSTFIELDS, $PostData);

		$result = curl_exec($ch);
		
		$cookie = $this->ParserCookie($result);
		$this->JSESSIONID = $cookie["JSESSIONID"];

		curl_close($ch);

		if(strpos($result,'Menu') !== FALSE){
			return TRUE;
		}else{
			return FALSE;
		}
	}

	function ParserCookie($result){
		preg_match_all('/^Set-Cookie:\s*([^;]*)/mi', $result, $matches);
		$cookies = array();
		foreach($matches[1] as $item) {
    		parse_str($item, $cookie);
    		$cookies = array_merge($cookies, $cookie);
		}
		return $cookies;
	}
}

// $aaa = new Attendance();
// if($aaa->Login("N125359408","820304")){
// 	$session = $aaa->GetJSESSIONID();
// 	// $aaa->DelRecord("AAAWcbAAIAAAd4NAAv",$session);
// 	// AAAWcbAAIAAAd4NAAv
// 	// $aaa->AddRecord("1060501","整理資料","105B1200",$session);
// 	// echo 1060428


// 	// echo $aaa->PrintRecord("105B1200","1060401","1060430",$session);

// 	// echo $aaa->GetRecord("N125359408",$session);

// 	// foreach($aaa->GetSchno($aaa->GetJSESSIONID()) as $i){
// 	// 	echo $i."\n";
// 	// }
// }
?>